firedrake/mesh.py

```diff
@@ -1080,8 +1081,6 @@ def init_cell_orientations(self, expr):
         import firedrake.function as function
         import firedrake.functionspace as functionspace
 
-        if expr.value_shape()[0] != 3:
-            raise NotImplementedError('Only implemented for 3-vectors')
         if self.ufl_cell() not in (ufl.Cell('triangle', 3),
                                    ufl.Cell("quadrilateral", 3),
                                    ufl.TensorProductCell(ufl.Cell('interval'), ufl.Cell('interval'),
```

Comment:
Does the UFL version now work for general dimension?  e.g. can you use it to call `init_cell_orientations` on a `CircleManifoldMesh`?